### PR TITLE
Improve documentation with React components

### PR DESCRIPTION
**Pull request by pepibumur**
### Short description 📝
I found the documentation a bit hard to navigate. This PR uses `mdx` to represent each of the manifest models with a table. Something like this:

image

### Solution 📦

The models are now represented with React components:

```js
<PropertiesTable props={[
{
  name: "Name",
  description: "The name of the target. The Xcode project target and the derivated product take the 
  same name.",
  type: "String",
  optional: false,
  default: ""
}]}/>
```

### CODE REVIEW

File: docs/usage/2-manifest.mdx [hunk 1]
```diff
@@ -98,47 +99,238 @@ default: "[]"
...
+let target = Target(name: "App",
+                    platform: .iOS,
+                    product: .app,
+                    bundleId: "io.tuist.App",
+                    infoPlist: "Info.plist",
+                    sources: "Sources/**",
+                    dependencies: [])
+```
+
+>>>>>>> Move more models to tables
```
Comment (kwridan) [inline docs/usage/2-manifest.mdx]: yikes looks like the conflict markers were accidentally added

File: docs/usage/2-manifest.mdx [hunk 2]
```diff
@@ -98,47 +99,238 @@ default: "[]"
...
 
-- **Dependencies (optional):** You can read more about dependencies [here](./dependencies.md)
-- **Settings (optional):** Read more about [settings](#settings)
-- **CoreDataModels (optional):** An array of `CoreDataModel` objects. A `CoreDataModel` is an special type of resource that requires the following two attributes:
-  - **Path:** Relative path to the Core Data model.
-  - **CurrentVersion:** Current version without the extension.
-- **Environment (optional):** It's a `[String: String]` to defined variables that will be set to the scheme that Tuist automatically generates for the target.
+<Message info title="Relative paths" description="Either the tool or the path need to be defined. Moreover, Tuist verifies whether the launch path is valid and fail otherwise."/>
 
 ### Scheme
```
Comment (kwridan) [inline docs/usage/2-manifest.mdx]: Is the idea to incrementally adopt the table for the various sections? e.g. Scheme still uses the list style

**Comment by pedropinerashopify** [comment]
@kwridan I [opened an issue](https://github.com/pedronauck/docz/issues/896) reporting the issue with the button on top of the document title when it's opened on mobile.

Regarding the layout on mobile, I was looking at the behaviour of the Table component that we are using from this [library](https://react.semantic-ui.com/collections/table/), and that's the default behaviour, which I agree is a bit odd. I can disable it but then the table would expand beyond the device viewport. Optionally, what I can do is presenting the attributes as a list if it's opened from a mobile device. What do you think @kwridan?

File: docs/usage/2-manifest.mdx (hunk 2 above)
Comment (pepibumur) [reply docs/usage/2-manifest.mdx]: I missed this one. It should be ported now to the new format 👍

**Comment by pepibumur** [comment]
@kwridan added some improvements:

* I improved the responsiveness of the tables when they are shown on a mobile device.

image

* Changed the style of the code blocks. Docz, the library that we are using for the documentation generation uses [CodeMirror](https://codemirror.net/demo/theme.html) as a code highlighter, which unfortunately can't parse the Swift code accurately. I tried to get as close as possible to the Xcode highlighting but it's a bit far from ideal IMO. Still better than the dark that we had before though.
* Changed the default font to use the ones that Apple is using on the Swift website. I think with that font the website looks fresher and certainly more readable.
 
I also ported the scheme's documentation to the new format using components. Would you mind having another look before we merge it?

**Comment by kwridan** [comment]
Ah it's a docz issue https://github.com/pedronauck/docz/issues/850

**Comment by pepibumur** [comment]
I fixed both, the headers and the scrolling issues. Moreover, I enabled a button at the top-right corner to edit the page on GitHub:

image

**Pull request merged**